src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/Statistics/BoundaryDataController.cs

```diff
@@ -0,0 +1,107 @@
+#nullable enable
+using GeoJSON.Net.Feature;
+using GovUk.Education.ExploreEducationStatistics.Admin.Requests;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels.Statistics;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Models.GlobalRoles;
+using Unit = GovUk.Education.ExploreEducationStatistics.Common.Model.Unit;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Controllers.Api.Statistics;
+
+[Route("api")]
+[ApiController]
+[Authorize(Roles = RoleNames.BauUser)]
+public class BoundaryDataController(
+    IBoundaryLevelService boundaryLevelService) : ControllerBase
+{
+    [HttpGet("boundary-data/levels")]
+    public async Task<ActionResult<List<BoundaryLevelViewModel>>> GetBoundaryLevels()
+    {
+        return await boundaryLevelService
+            .ListBoundaryLevels()
+            .OnSuccess(boundaryLevels => boundaryLevels.Select(BuildViewModel).ToList())
+            .HandleFailuresOrOk();
+    }
+
+    [HttpGet("boundary-data/levels/{id:long}")]
+    public async Task<ActionResult<BoundaryLevelViewModel>> GetBoundaryLevel(long id)
+    {
+        return await boundaryLevelService
+            .GetBoundaryLevel(id)
+            .OnSuccess(BuildViewModel)
+            .HandleFailuresOrOk();
+    }
+
+    [HttpPatch("boundary-data/levels")]
+    public async Task<ActionResult> UpdateBoundaryLevel(BoundaryLevelUpdateRequest updateRequest)
+    {
+        return await boundaryLevelService
+            .UpdateBoundaryLevel(updateRequest.Id, updateRequest.Label)
+            .HandleFailuresOrNoContent();
+    }
+
+    [HttpPost("boundary-data")]
+    public async Task<ActionResult<Unit>> CreateBoundaryLevel(
+        [FromForm] GeographicLevel level,
+        [FromForm] string label,
+        IFormFile file,
+        [FromForm] DateTime published)
+    {
+        return await boundaryLevelService
+            .ListBoundaryLevels()
+            .OnSuccessDo(async existingLevels =>
+            {
+                if (file.FileName.Split('.')[1] != "geojson")
```

Comment:
I've moved file name and label validation to a FluentValidation class.

Regarding the view model creation, I moved this function from where it was before based on another comment. I wasn't happy moving it to a controller because the presentation layer now has a reference to a database entity. I can see that other places have view model creation in service layers (which I'm quite sure is where I put it before, imo where it should be), but essentially there doesn't appear to be a consistent approach atm so I'm not moving it again :D.